domain-models-interface-extensions/src/main/java/org/folio/rest/tools/annotator/CustomArrayAnnotator.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.rest.tools.annotator;
+
+import java.util.Optional;
+
+import org.folio.rest.annotations.ElementsPattern;
+import org.folio.rest.annotations.NoNullElements;
+import org.jsonschema2pojo.AbstractAnnotator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.sun.codemodel.JDefinedClass;
+import com.sun.codemodel.JFieldVar;
+
+public class CustomArrayAnnotator extends AbstractAnnotator {
+
+  @Override
+  public void propertyField(JFieldVar field, JDefinedClass clazz, String propertyName, JsonNode propertyNode) {
+    super.propertyField(field, clazz, propertyName, propertyNode);
+    Optional<JsonNode> typeNode = Optional.ofNullable(propertyNode.get("type"));
+    if (typeNode.isPresent() && typeNode.get().asText().equals("array")) {
+      field.annotate(NoNullElements.class);
+      Optional<JsonNode> itemsNode = Optional.ofNullable(propertyNode.get("items"));
+      if (itemsNode.isPresent()) {
+        Optional<JsonNode> itemsTypeNode = Optional.ofNullable(itemsNode.get().get("type"));
+        Optional<JsonNode> patternNode = Optional.ofNullable(itemsNode.get().get("pattern"));
+        if (itemsTypeNode.isPresent() && itemsTypeNode.get().asText().equals("string") && patternNode.isPresent()) {
+          field.annotate(ElementsPattern.class).param("regexp", patternNode.get().asText());
+        }
+      }
+    }
+  }
```

Comment:
Doing https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html , dropping Optional and moving the null test into a function gives the code a different look (and more lines): 
```
  private String text(JsonNode node) {
    if (node == null) {
      return null;
    }
    return node.asText();
  }

  @Override
  public void propertyField(JFieldVar field, JDefinedClass clazz, String propertyName, JsonNode propertyNode) {
    super.propertyField(field, clazz, propertyName, propertyNode);
    JsonNode typeNode = propertyNode.get("type");
    if (! "array".equals(text(typeNode))) {
      return;
    }
    field.annotate(NoNullElements.class);
    JsonNode itemsNode = propertyNode.get("items");
    if (itemsNode == null) {
      return;
    }
    JsonNode itemsTypeNode = itemsNode.get("type");
    if (! "string".equals(text(itemsTypeNode))) {
      return;
    }
    String pattern = text(itemsNode.get("pattern"));
    if (pattern == null) {
      return;
    }
    field.annotate(ElementsPattern.class).param("regexp", pattern);
  }
```